nexus/src/db/model/macaddr.rs

```diff
@@ -8,55 +8,87 @@ use diesel::pg::Pg;
 use diesel::serialize::{self, ToSql};
 use diesel::sql_types;
 use omicron_common::api::external;
-use rand::{rngs::StdRng, SeedableRng};
-use std::convert::TryFrom;
+use rand::thread_rng;
+use rand::Rng;
 
 #[derive(Clone, Copy, Debug, PartialEq, AsExpression, FromSqlRow)]
-#[diesel(sql_type = sql_types::Text)]
+#[diesel(sql_type = sql_types::BigInt)]
 pub struct MacAddr(pub external::MacAddr);
 
 impl MacAddr {
-    /// Generate a unique MAC address for an interface
-    pub fn new() -> Result<Self, external::Error> {
-        use rand::Fill;
-        // Use the Oxide OUI A8 40 25
-        let mut addr = [0xA8, 0x40, 0x25, 0x00, 0x00, 0x00];
-        addr[3..].try_fill(&mut StdRng::from_entropy()).map_err(|_| {
-            external::Error::internal_error("failed to generate MAC")
-        })?;
-        // From RFD 174, Oxide virtual MACs are constrained to have these bits
-        // set.
-        addr[3] |= 0xF0;
-        // TODO-correctness: We should use an explicit allocator for the MACs
-        // given the small address space. Right now creation requests may fail
-        // due to MAC collision, especially given the 20-bit space.
-        Ok(Self(external::MacAddr(macaddr::MacAddr6::from(addr))))
+    // Guest MAC addresses begin with the Oxide OUI A8:40:25. Further, guest
+    // address are constrained to be in the virtual address range
+    // A8:40:24:F_:__:__. Even further, the range F0:00:00 - FE:FF:FF is
+    // reserved for customer-visible addresses (FF:00:00-FF:FF:FF is for
+    // system MAC addresses). See RFD 174 for the discussion of the virtual
+    // range, and
+    // https://github.com/oxidecomputer/omicron/pull/955#discussion_r856432498
+    // for an initial discussion of the customer/system address range split.
+    pub(crate) const MIN_GUEST_ADDR: i64 = 0xA8_40_25_F0_00_00;
+    pub(crate) const MAX_GUEST_ADDR: i64 = 0xA8_40_25_FE_FF_FF;
+
+    /// Generate a random MAC address for a guest network interface
+    pub fn random_guest() -> Self {
+        let value =
+            thread_rng().gen_range(Self::MIN_GUEST_ADDR..=Self::MAX_GUEST_ADDR);
+        Self::from_i64(value)
+    }
+
+    /// Construct a MAC address from its i64 big-endian byte representation.
+    // NOTE: This is the representation used in the database.
```

Comment:
What's the reason for storing the value in big endian?

(It's not really a problem, it just doesn't seem necessary to me - AFAICT, we're storing nearly all other values in the DB as host order / little endian)